### ISSUE TITLE
Assignment revision history incorrectly includes merge commit (may affect late penalty?)

### ISSUE THREAD

**Issue by falsifian** [MarkUsProject/Markus#4534]
This bug is based on the following sequence of events:

* I pull student's git repo from MarkUs.
* Student uploads assignment A, producing commit C1.
* I make a change* to an unrelated assignment B, producing commit C2.
* I merge C1 and C2 to produce C3, and push that to MarkUs.

The repository view `https://markus.teach.cs.toronto.edu/csc240-2020-01/assignments/XX/submissions/XX/repo_browser` now has two problems:

* The menu beside "Viewing Revision" shows C3 as the latest commit.
* the "Revised by" column shows my name (the instructor) instead of theirs, presumably because of C3.

The student emailed me, concerned that they would get a late penalty because they see the date from C3.

If I run `git log <path to assignment A file>', git shows commit C1, and not C2 or C3. So it seems like `git log` does the right thing here. I don't know how MarkUs is doing it, but at least in this case I think it's not correct.

*(If you're curious why I was fiddling with a student's assignment: I was converting a quiz they uploaded to pdf since MarkUs handles pdfs better than images.)

**Comment by falsifian** [comment]
I confirmed that MarkUs indeed added a late penalty for the student (which I removed manually).

**Comment by mishaschwartz** [comment]
Hi @falsifian, just to confirm. You're saying:

- you cloned the remote repo prior to C1 being created
- then the student committed and pushed C1 to the remote repo
- then you merged the changes from the remote repo (including C1) to your local clone creating C3 and pushed the resulting commits back to the remote

Also just to confirm:

- all commits were made to the master branch (no other branches were created)
- when you merged the remote changes to the local branch you did or did not use the `--squash` flag?

**Comment by falsifian** [comment]
> Hi @falsifian, just to confirm. You're saying:
> 
>     * you cloned the remote repo prior to C1 being created
> 
>     * then the student committed and pushed C1 to the remote repo
> 
>     * then you merged the changes from the remote repo (including C1) to your local clone creating C3 and pushed the resulting commits back to the remote

Yes. (I cloned long ago and my most recent pull was before C1.) Here's a piece of the output of `git log --graph` with the student ID removed:

    *   commit 98bd3798178df047dc804a26f47da84ef85e2e76 (HEAD -> master, origin/master, origin/HEAD)
    |\  Merge: 2ec112f 1addc07
    | | Author: James Cook <jcook@cs.berkeley.edu>
    | | Date:   Tue Apr 14 05:03:33 2020 +0000
    | | 
    | |     Merge branch 'master' of https://markus.teach.cs.toronto.edu/git/csc240-2020-01/XXX
    | | 
    | * commit 1addc07237db824cc0de2d7ee2a9f1328e38c3c3
    | | Author: XXX
    | | Date:   Tue Apr 14 00:49:24 2020 -0400
    | | 
    | |     Update files
    | | 
    | * commit 0ae4d3157419ed9d420c4cfd7f8a65e6926e7435
    | | Author: XXX
    | | Date:   Tue Apr 14 00:45:48 2020 -0400
    | | 
    | |     Update files
    | | 
    * | commit 2ec112f40155c9122c20892f5510015436b73c32
    |/  Author: James Cook <jcook@cs.berkeley.edu>
    |   Date:   Tue Apr 14 05:02:55 2020 +0000
    |   
    |       Instructor commit: convert to pdf.
    | 
    * commit 602aa0f4d3b37b956c13b39157820d3e9d84cba4


> 
> Also just to confirm:
> 
>     * all commits were made to the master branch (no other branches were created)
> 
>     * when you merged the remote changes to the local branch you did or did not use the `--squash` flag?

Right. I never created any branches and I didn't use `--squash`.

I believe the same thing happened with a second student (I had a similar merge commit and their assignment was similarly marked late). I haven't confirmed (didn't look too closely) but could if it would help.

**Comment by mishaschwartz** [comment]
Ok thank you for the additional info. That is really helpful and will help me to recreate the issue. I'll look into it.
